sqlmesh/core/snapshot/evaluator.py

```diff
@@ -767,21 +761,25 @@ def demote(
 class MaterializableStrategy(PromotableStrategy):
     def append(
         self,
-        model: Model,
+        snapshot: Snapshot,
         table_name: str,
         query_or_df: QueryOrDF,
         snapshots: t.Dict[str, Snapshot],
         is_dev: bool,
         **kwargs: t.Any,
     ) -> None:
+        model = snapshot.model
         self.adapter.insert_append(table_name, query_or_df, columns_to_types=model.columns_to_types)
 
     def create(
         self,
-        model: Model,
+        snapshot: Snapshot,
         name: str,
         **render_kwargs: t.Any,
     ) -> None:
+        model = snapshot.model
+        self.adapter.execute(model.render_pre_statements(**render_kwargs))
```

Comment:
Why did we move this for `create` but not for `evaluate`?

Reply:
What if `create` is overridden in one of the strategies, do they also need to invoke pre / post statements too?

Reply:
I'll move it back for `create` and early return if not a `model` like done in `evaluate`.